webui/react/src/pages/Templates/TemplatesList.tsx

```diff
@@ -0,0 +1,289 @@
+        multiple
+        values={settings.workspace?.map((ws) => ws.toString())}
+        width={220}
+        onFilter={handleWorkspaceFilterApply}
+        onReset={handleWorkspaceFilterReset}
+      />
+    ),
+    [handleWorkspaceFilterApply, handleWorkspaceFilterReset, settings.workspace],
+  );
+
+  const handleDropdown = useCallback(
+    (key: string, record: Template) => {
+      switch (key) {
+        case MenuKey.ViewTemplate:
+          setSelectedTemplate(record);
+          TemplateViewModal.open();
+          break;
+      }
+    },
+    [TemplateViewModal],
+  );
+
+  const handleNameSearchReset = useCallback(() => {
+    updateSettings({ name: undefined });
+  }, [updateSettings]);
+
+  const handleNameSearchApply = useCallback(
+    (newSearch: string) => {
+      updateSettings({ name: newSearch || undefined, tableOffset: 0 });
+    },
+    [updateSettings],
+  );
+
+  const nameFilterSearch = useCallback(
+    (filterProps: FilterDropdownProps) => (
+      <TableFilterSearch
+        {...filterProps}
+        value={settings.name || ''}
+        onReset={handleNameSearchReset}
+        onSearch={handleNameSearchApply}
+      />
+    ),
+    [handleNameSearchApply, handleNameSearchReset, settings.name],
+  );
+
+  const columns = useMemo(() => {
+    const actionRenderer = (_: string, record: Template) => (
+      <Dropdown menu={DROPDOWN_MENU} onClick={(key) => handleDropdown(key, record)}>
+        <Button icon={<Icon name="overflow-vertical" title="Action menu" />} type="text" />
+      </Dropdown>
+    );
+
+    return [
+      {
+        dataIndex: 'name',
+        defaultWidth: DEFAULT_COLUMN_WIDTHS['name'],
+        filterDropdown: nameFilterSearch,
+        filterIcon: <Icon name="search" size="tiny" title="Search" />,
+        isFiltered: (settings: Settings) => !!settings.name,
+        key: 'name',
+        sorter: (a: Template, b: Template): number => alphaNumericSorter(a.name, b.name),
+        title: 'Name',
+      },
+      {
+        align: 'center',
+        dataIndex: 'workspace',
+        defaultWidth: DEFAULT_COLUMN_WIDTHS['workspace'],
+        filterDropdown: workspaceId ? undefined : workspaceFilterDropdown,
+        filters: workspaceId
+          ? undefined
+          : workspaces.map((ws) => ({
+              text: <WorkspaceFilter workspace={ws} />,
+              value: ws.id,
+            })),
+        isFiltered: (settings: Settings) => !!settings.workspace?.length,
+        key: V1GetTemplatesRequestSortBy.NAME,
+        render: (_v: string, record: Template) => taskWorkspaceRenderer(record, workspaces),
+        title: 'Workspace',
+      },
+      {
+        align: 'right',
+        className: 'fullCell',
+        dataIndex: 'action',
+        defaultWidth: DEFAULT_COLUMN_WIDTHS['action'],
+        fixed: 'right',
+        key: 'action',
+        render: actionRenderer,
+        title: '',
+        width: DEFAULT_COLUMN_WIDTHS['action'],
+      },
+    ] as ColumnDef<Template>[];
```

Comment:
**question:** would it work to have `const columns: ColumnDef<Template>[]`?

Reply:
The `isFiltered` fields are not compatible because we only take `(s: unknown) => boolean`, but want to pass `(s: Settings) => boolean;`. I'm not sure why we only accept `unknown` as parameter though...